### ISSUE TITLE
sync sources is not removing the deleted table from database - UI for T

### ISSUE THREAD

**Issue by siddhant3030** [DalgoT4D/webapp#1014]
For CMHLP 

I just deleted one of the table from the database which was not needed and did the sync sources again in the UI for T which did not remove the model from UI for T. is this expected? It should get removed right?

<img width="1238" alt="Screenshot 2024-07-17 at 12 05 46 PM" src="https://github.com/user-attachments/assets/0a1fe0d2-ad9d-4cde-9396-84ebc35cef75">

**Comment by fatchat** [comment]
yes it should get removed, correct

**Comment by fatchat** [comment]
the issue here is that if a source table gets deleted in the warehouse BUT is also part of a pipeline, we should not delete that source from our source list

possible solution: allow the user to delete a source from the tree view, warning them etc etc

cc: @nairabhishek73

**Comment by fatchat** [comment]
more generally we need a decision on when we can delete sources/models at our end when something disappears from the client's warehouse

**Comment by nairabhishek73** [comment]
We are currently working with the assumption that only the highest level/s of permission (and competence) within the org have access to the database/warehouse, so if they delete something is it Dalgo's responsibility to deal with this error? 

When deleting tables from the database what notifications/warnings do you get? Don't want to double the warnings if its warning you there.

If we notify users on the platform and they realise a mistake has been made then can they restore the table to their database? How? That would be an ideal solution if this problem came up.

**Comment by fatchat** [comment]
if they delete a table in their warehouse it is not dalgo's responsibility no

dalgo can detect that a table has been deleted, and in those cases we can notify them and ask whether we should remove our reference to that table or not

whether or not they can restore a deleted table will depend on their database person / team and will vary by setup

**Comment by nairabhishek73** [comment]
Right, some source table deleted, next pipeline run fails, intermediate tables that have been created previously will still be available in the warehouse. (correct me if i'm wrong here)

Re: Dalgo can detect that a table has been deleted, and in those cases we can notify them and ask whether we should remove our reference to that table or not- This will be helpful. 

Need to think through what it should look like if this table is linked to more intermediate tables and production tables downstream.
